pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -284,69 +305,84 @@ func ValidateServiceaccountMapping(p *v1beta1.PipelineSpec, pr *v1beta1.Pipeline
 // the spec. If it is unable to retrieve an instance of a referenced Task, it  will return
 // an error, otherwise it returns a list of all of the Tasks retrieved.  It will retrieve
 // the Resources needed for the TaskRun using the mapping of providedResources.
+//
+// TODO: The proliferation of Get* methods here is a smell. This method should
+// take a versioned clientset instead which includes methods to retrieve these
+// resources, with generated fakes for testing.
```

Comment:
I agree with the start of the comment (it's a smell), not with the rest though (the *solution*). "versioned clientset" assumes the `get` methods only use k8s clients, which is not the case (with Tekton OCI bundles, we do fetch through a very different method :upside_down_face:)

Reply:
This is Jason's comment.  @ImJasonH should I change or remove the comment?

Reply:
I'm fine to remove it. I agree with Vincent that with the addition of OCI bundles the prescribed remedy for the smell might not be correct.